### PR TITLE
anomaly-detector

### PR DESCRIPTION
**Pull request by csumpter**
Add support for anomaly detector resource.

> Please ensure new code produces a valid CloudFormation template.

Checked and verified.

Cloudformation docs on this resource can be found here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-cloudwatch-anomalydetector.html

### CODE REVIEW

File: src/crucible/aws/cloudwatch/anomaly_detector.clj [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+            [crucible.encoding.keys :refer [->key]]
+            [crucible.resources :refer [defresource spec-or-ref]]))
+
+(s/def ::value (spec-or-ref string?))
+
+(s/def ::name (spec-or-ref string?))
+
+(s/def ::dimension (s/keys :req [::name ::value]))
+
+(s/def ::dimensions (s/coll-of ::dimension :kind vector?))
```
Comment (shooit) [inline src/crucible/aws/cloudwatch/anomaly_detector.clj]: Does this need to be a vector? The JSON serializer will coerce non-map collections to an array anyway

File: src/crucible/aws/cloudwatch/anomaly_detector.clj [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+(s/def ::end-time ::date-time)
+
+(s/def ::start-time ::date-time)
+
+(s/def ::range (s/keys :req [::start-time ::end-time]))
+
+(s/def ::ranges (s/coll-of ::range))
+
+(s/def ::name-space (spec-or-ref string?))
```
Comment (shooit) [inline src/crucible/aws/cloudwatch/anomaly_detector.clj]: `namespace` is a happy compound word in clojure. Then you wouldn't have to `->key` it.

File: src/crucible/aws/cloudwatch/anomaly_detector.clj (hunk 1 above)
Comment (csumpter) [reply src/crucible/aws/cloudwatch/anomaly_detector.clj]: I have removed the requirement from the spec: https://github.com/brabster/crucible/pull/188/commits/73a771b0cbdefd96d6d4ca010eb8eaeccdd4321a

File: src/crucible/aws/cloudwatch/anomaly_detector.clj (hunk 2 above)
Comment (csumpter) [reply src/crucible/aws/cloudwatch/anomaly_detector.clj]: Feedback addressed in https://github.com/brabster/crucible/pull/188/commits/1c698299a9b9a2768de0b59360f37a07cc0186df

**Pull request merged**